src/main/java/io/vertx/core/http/impl/HttpClientImpl.java

```diff
@@ -710,6 +713,24 @@ private HttpClientRequest doRequest(HttpMethod method, String host, int port, St
     Objects.requireNonNull(host, "no null host accepted");
     Objects.requireNonNull(relativeURI, "no null relativeURI accepted");
     checkClosed();
+    ProxyOptions proxyOptions = options.getProxyOptions();
+    if (!options.isSsl() && proxyOptions != null && proxyOptions.getType() == ProxyType.HTTP) {
+      if (headers == null) {
+        headers = MultiMap.caseInsensitiveMultiMap();
+      }
+      relativeURI = "http://" + host + (port != 80 ? ":" + port : "") + relativeURI;
+      host = proxyOptions.getHost();
+      port = proxyOptions.getPort();
+      log.debug("changing request to proxy request " + relativeURI);
+      log.debug("username "+proxyOptions.getUsername()+" password "+proxyOptions.getPassword());
+      if (proxyOptions.getUsername() != null && proxyOptions.getPassword() != null) {
+        log.debug("adding authorization header");
+        headers.add("Proxy-Authorization", "Basic " + Base64.getEncoder()
+            .encodeToString((proxyOptions.getUsername() + ":" + proxyOptions.getPassword()).getBytes()));
+      }
+      // TODO: setting the host here does not work yet
+      headers.add("Host", host + (port != 80 ? ":" + port : ""));
```

Comment:
what is the purpose of this line ? normally the host header is set at connect time because it's not the same for HTTP and HTTP/2 using HttpClientRequestImpl hostHeader() method